main.py

```diff
@@ -71,6 +74,32 @@ def on_keypress(bit_index):
       alive_enemies.remove(enemy)
     else:
       enemy.update_position()
+      
+  if len(missiles_to_shoot) > 0:
+    for missiles in missiles_to_shoot:
+      if missiles.has_collided:
+        missiles.enemy.is_destroyed = True
+        missiles.enemy.update_position()
+        missiles_to_shoot.remove(missiles)
+        continue
+      if len(missiles.missiles_in_place) == len(missiles.missiles):
+        missiles.is_shot = True
+      missiles.update_locations()
+
+  for enemy in alive_enemies:
+    if not enemy.being_destroyed:
+      if binary_bar_preview.current_hexadecimals == enemy.current_hexadecimals:
+        def check_flip(acc, box):
+          if box.is_flipped:
+            box.flip_bit()
+            return [*acc, box.missile.vertices]
+          return acc
+
+        active_missile_locations = reduce(check_flip, binary_boxes, [])
```

Comment:
While I took the opportunity to introduce functional iterations, a reduction here really doesn't provide much benefit for the bit of extra complexity. 
If there wasn't the `box.flip_bit()` call I would have said a list comprehension should definitely be used, but since it is there, building up the new list with a basic for loop is still perfectly valid.
So you don't need to keep the reduction if you don't want to, but nothing wrong with leaving it either.